cuda/Simulation.cu

```diff
@@ -23,27 +26,22 @@ unsigned long long run_event_based_simulation_baseline(Inputs in, SimulationData
 	int nthreads = 256;
 	int nblocks = ceil( (double) in.lookups / (double) nthreads);
 
+
 	xs_lookup_kernel_baseline<<<nblocks, nthreads>>>( in, GSD );
 	gpuErrchk( cudaPeekAtLastError() );
-	gpuErrchk( cudaDeviceSynchronize() );
-
-    size_t sz = in.lookups * sizeof(unsigned long);
-	unsigned long * v = (unsigned long *) malloc(sz);
-	gpuErrchk( cudaMemcpy(v, GSD.verification, sz, cudaMemcpyDeviceToHost) );
-
-#ifdef ALIGNED_WORK
-	*end = get_time();
-#endif
-
+	
 	////////////////////////////////////////////////////////////////////////////////
 	// Reduce Verification Results
 	////////////////////////////////////////////////////////////////////////////////
 
 	if( mype == 0)	printf("Reducing verification results...\n");
+    gpuErrchk(cudaMemcpy(SD.verification, GSD.verification, in.lookups * sizeof(unsigned long), cudaMemcpyDeviceToHost) );
+
+    unsigned long verification_scalar = 0;
+    for( int i =0; i < in.lookups; i++ )
+        verification_scalar += SD.verification[i];
```

Comment:
Indentation trouble, should use 8-wide tab instead of four spaces here and throughout.